### PR TITLE
Post spam time protections

### PR DESCRIPTION
**Pull request by inferno759**
created spam detection service for posts and method for getting recent user's posts. Spam Detection should have a timeout to set for intervals that posts are allowed to be made, and disallows posts with the exact same content as previous posts in the last X minutes.

### CODE REVIEW

**Comment by inferno759** [comment]
Broke CheckSpam into multiple methods, added CheckSpamService into the Controller integration test for adding posts. Added 4 tests for testing correct and incorrect Post spam by comparing recent post time, and same content.

File: Fakebook.Posts/Fakebook.Posts.RestApi/Services/CheckSpamService.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+			_postRepository = postRepository;
+			_timeService = timeService;
+		}
+
+		public async Task<bool> IsPostNotSpam(Post userPost)
+		{
+			if(userPost.Content == null)
+            {
+				return false;
+            }
```
Comment (escalonn) [inline Fakebook.Posts/Fakebook.Posts.RestApi/Services/CheckSpamService.cs:27]: format this file, the whitespace is all over the place.

File: Fakebook.Posts/Fakebook.Posts.UnitTests/Services/SpamServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+            oldPosts.Add(post);
+
+            bool isNotSpam = checkSpamService.CheckTimeSpam(oldPosts, secondsTimeout, dateNow);
+            Assert.True(isNotSpam);
+        }
+
+        [Fact]
+        public void CheckSpamService_IsNotSpammingByTime_IsFalse()
+        {
+
```
Comment (escalonn) [inline Fakebook.Posts/Fakebook.Posts.UnitTests/Services/SpamServiceTests.cs:57]: empty lines don't belong right next to `{` and `}`. but i'm not going to insist on it now

File: Fakebook.Posts/Fakebook.Posts.RestApi/Services/CheckSpamService.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                {
+					isNotSpam = false;
+					return isNotSpam;
+                }
+            }
+			isNotSpam = true;
+			return isNotSpam;
+        }
+
+		public bool CheckSamePostSpam(List<Post> posts, string userContent)
```
Comment (escalonn) [inline Fakebook.Posts/Fakebook.Posts.RestApi/Services/CheckSpamService.cs:68]: method needs to be documented (in the interface also), the return value meaning is not obvious.

File: Fakebook.Posts/Fakebook.Posts.DataAccess/Repositories/PostsRepository.cs [hunk 4]
```diff
@@ -144,6 +144,23 @@ public IEnumerator<Post> GetEnumerator()
...
+            var query = await _context.Posts
+                .Where(u => u.UserEmail == userEmail)
+                .Where(t => DateTimeOffset.Compare(t.CreatedAt, latestTime) >= 0)
+                .ToListAsync();
+            var queryResult = query.Select(s => s.ToDomain());
+            return queryResult.ToList();
+        }
+        
+
+
```
Comment (escalonn) [inline Fakebook.Posts/Fakebook.Posts.DataAccess/Repositories/PostsRepository.cs:163]: why so many random empty lines added all over this PR? it's just sloppy. i'll let it through except for that one file

**Review by escalonn — CHANGES_REQUESTED** [review]

File: Fakebook.Posts/Fakebook.Posts.DataAccess/Repositories/PostsRepository.cs (hunk 4 above)
Comment (inferno759) [reply Fakebook.Posts/Fakebook.Posts.DataAccess/Repositories/PostsRepository.cs:163]: ok :(

**Review by escalonn — APPROVED** [review]

File: Fakebook.Posts/Fakebook.Posts.RestApi/Services/CheckSpamService.cs (hunk 1 above)
Comment (inferno759) [reply Fakebook.Posts/Fakebook.Posts.RestApi/Services/CheckSpamService.cs]: for some reason, github is showing the brackets as far left when they are not.

Comment (escalonn) [reply Fakebook.Posts/Fakebook.Posts.RestApi/Services/CheckSpamService.cs]: it's because you're mixing tabs and spaces, and github shows tabs with width 8

**Review by escalonn — CHANGES_REQUESTED** [review]
no tabs in indentation

**Comment by nosyrbllewe** [comment]
> no tabs in indentation

Is this approved then or not?

**Review by escalonn — APPROVED** [review]
the checkspamservice file is still full of mixed tabs and spaces. ordinarily it would definitely not be an approval, but we're low on time so i'll leave it to your call

**Pull request merged**